extensions/amp-story/0.1/amp-story.js

```diff
@@ -1501,18 +1502,27 @@ export class AmpStory extends AMP.BaseElement {
     const nextPageId = currentPage
         .getNextPageId(true /*opt_isAutomaticAdvance */);
 
-
     if (nextPageId) {
+      const nextPage = this.getPageById_(nextPageId);
+
+      // we do not ever want two consecutive ads
```

Comment:
can we have this logic in auto-ads?

Reply:
Short answer: not without figuring out the reverse dependency problem we had before. 

Right now `auto-ads` doesn't know about the existing pages array. It only has access to the `pageId` of the page that triggered the event. It knows what ad pages have been created (and their ids etc) but not where they have been placed.

I know, we need a `PageManager` 😬

Reply:
We need to expose `getCurrentPage` & `getNextPage` methods from amp-story.js, which later on can be moved to the `PageManager`. 

the below logic should not belong to either amp-story.js or page-manager.js
```
currentPage.isAd() || nextPage.isAd()
```

Reply:
moved to `auto-ad`